### PR TITLE
feat: increase the robustness of megaparse

### PR DESCRIPTION
**Pull request by jacopo-chevallard**
- passing base url and timeout explicitly, so that they can be configured from the module caling the sdk 
- improving the way we check for supported models, which are now defined in a separate module
- using docker-compose instead of a Dockerfile directly

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -45,5 +45,3 @@ ENV PYTHONPATH="/app:/app/megaparse/sdk"
 
 COPY . .
 EXPOSE 8000
-
-CMD ["uvicorn", "megaparse.api.app:app", "--host", "0.0.0.0", "--port", "8000"]
```
Comment (StanGirard) [inline Dockerfile:49]: don't remove that or docker can't start unless provided explicit command

File: megaparse/sdk/megaparse_sdk/__init__.py [hunk 2]
```diff
@@ -4,8 +4,8 @@
 
 
 class MegaParseSDK:
-    def __init__(self, api_key: str):
-        self.client = MegaParseClient(api_key)
+    def __init__(self, api_key: str, base_url: str | None = None, timeout: int = 600):
```
Comment (StanGirard) [inline megaparse/sdk/megaparse_sdk/__init__.py:7]: Timeout won't affect the http request. By default in production, the timeout is on the laod balancer and not the app. It is at 60 seconds. Adding this here gives a false sense of: you can increase the timeout

File: Dockerfile (hunk 1 above)
Comment (jacopo-chevallard) [reply Dockerfile:49]: Ok, but we don't have this in quivr-core for instance

File: megaparse/sdk/megaparse_sdk/__init__.py (hunk 2 above)
Comment (jacopo-chevallard) [reply megaparse/sdk/megaparse_sdk/__init__.py:7]: (and this is how I fixed it locally, but I understand that this is not a good solution for prod...)

File: docker-compose.dev.yml
Comment (chloedia) [inline docker-compose.dev.yml:1]: Why do we need docker compose in the repo?

File: megaparse/sdk/megaparse_sdk/client.py [hunk 3]
```diff
@@ -1,22 +1,23 @@
...
 class MegaParseClient:
-    def __init__(self, api_key: str | None = None):
-        self.base_url = os.getenv(
-            "MEGAPARSE_URL", "https://megaparse.tooling.quivr.app"
-        )
-
+    def __init__(
+        self,
+        api_key: str | None = None,
+        base_url: str | None = None,
```
Comment (chloedia) [inline megaparse/sdk/megaparse_sdk/client.py:10]: I like those in env variables but as you want

**Review by chloedia — APPROVED** [review]
Nice PR

File: megaparse/core/parser/entity.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    """Possible tags for the elements in the file"""
+
+    TABLE = "TABLE"
+    TOC = "TOC"
+    HEADER = "HEADER"
+    IMAGE = "IMAGE"
+
+
+class SupportedModel(Enum):
+    GPT_4O = ("gpt-4o", None)
```
Comment (AmineDiro) [inline megaparse/core/parser/entity.py:15]: Thats Nice ! 

Additionnal typeing can be used here  (either NamedTuples or typped dict ...)

File: megaparse/sdk/megaparse_sdk/__init__.py (hunk 2 above)
Comment (AmineDiro) [reply megaparse/sdk/megaparse_sdk/__init__.py:7]: Agree with @StanGirard  👍🏼 . We should have faster parsing always  or send the parser a page by page of the pdf from the workers

**Review by AmineDiro — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (jacopo-chevallard) [reply Dockerfile:49]: fixed

**Pull request merged**